packages/abstract-core/src/objects/asset_entry.rs

```diff
@@ -26,22 +28,56 @@ impl AssetEntry {
     /// Retrieve the source chain of the asset
     /// Example: osmosis>juno>crab returns osmosis
     /// Returns string to remain consistent with [`Self::asset_name`]
-    pub fn src_chain(&self) -> String {
-        self.0
-            .split(CHAIN_DELIMITER)
-            .next()
-            .unwrap_or("")
-            .to_string()
+    pub fn src_chain(&self) -> AbstractResult<String> {
+        let mut split = self.0.splitn(2, CHAIN_DELIMITER);
+
+        match split.next() {
+            Some(src_chain) => {
+                if src_chain.is_empty() {
+                    return self.entry_formatting_error();
+                }
+                // Ensure there's at least one more element (asset_name)
+                let maybe_asset_name = split.next();
+                if maybe_asset_name.is_some() && maybe_asset_name != Some("") {
+                    Ok(src_chain.to_string())
+                } else {
+                    self.entry_formatting_error()
+                }
+            }
+            None => self.entry_formatting_error(),
+        }
     }
 
     /// Retrieve the asset name without the src chain
     /// Example: osmosis>juno>crab returns juno>crab
-    pub fn asset_name(&self) -> String {
-        self.0
-            .split(CHAIN_DELIMITER)
-            .skip(1)
-            .collect::<Vec<&str>>()
-            .join(CHAIN_DELIMITER)
+    pub fn asset_name(&self) -> AbstractResult<String> {
```

Comment:
When would a user want to use this? It's unsafe for them to resolve because it will return a different assets so this conversion could be dangerous.